lib/ansible/module_utils/facts/network/linux.py

```diff
@@ -147,9 +147,13 @@ def get_interfaces_info(self, ip_path, default_ipv4, default_ipv6):
                 interfaces[device]['perm_macaddress'] = get_file_content(os.path.join(path, 'bonding_slave', 'perm_hwaddr'), default='')
             if os.path.exists(os.path.join(path, 'device')):
                 interfaces[device]['pciid'] = os.path.basename(os.readlink(os.path.join(path, 'device')))
+            if os.path.exists(os.path.join(path, 'duplex')):
+                duplex = get_file_content(os.path.join(path, 'duplex'))
+                if duplex is not None and duplex != 'unknown':
```

Comment:
hi, @alikins 
In my case, when the switch is unconfigured or configured wrong,  or there is a problem with the cable and can not communicate properly. It will display "unknown".
When you search google with keyword "Speed: Unknown! Duplex: Unknown!" you will get some example from others.